pkg/server/misc.go

```diff
@@ -0,0 +1,751 @@
+			}
+		}
+		percentage := float64(index) / float64(len(namespaces)) * BundleProgressPercentageLogs
+		sb.ProgressPercentage = BundleProgressPercentageYaml + int(percentage)
+	}
+}
+
+func streamLogToFile(logStream io.ReadCloser, path string, errLog io.Writer) {
+	var err error
+	defer func() {
+		if err != nil {
+			fmt.Fprintf(errLog, "Support Bundle: failed to generate %v: %v\n", path, err)
+		}
+	}()
+	err = os.MkdirAll(filepath.Dir(path), os.FileMode(0755))
+	if err != nil {
+		return
+	}
+	f, err := os.Create(path)
+	if err != nil {
+		return
+	}
+	defer f.Close()
+	_, err = io.Copy(f, logStream)
+	if err != nil {
+		return
+	}
+}
+
+func (s *HarvesterServer) getExternalSupportBundles(bundleMeta *BundleMeta, toDir string, errLog io.Writer) {
+	var err error
+	defer func() {
+		if err != nil {
+			fmt.Fprintf(errLog, "Support Bundle: failed to get external bundle: %v\n", err)
+		}
+	}()
+	err = os.Mkdir(toDir, os.FileMode(0755))
+	if err != nil {
+		return
+	}
+
+	lhBackend := fmt.Sprintf("http://%s.%s:%d", LonghornBackendServiceName, LonghornSystemNameSpace, LonghornBackendServicePort)
+	lhSbMgr := sb.NewLonghornSupportBundleManager(s.Context, lhBackend)
+	err = lhSbMgr.GetLonghornSupportBundle(bundleMeta.IssueURL, bundleMeta.IssueDescription, toDir)
+	if err != nil {
+		return
+	}
+}
+
+func (s *HarvesterServer) getNodeSupportBundles(bundleMeta *BundleMeta, toDir string, errLog io.Writer) {
+	err := s.collectNodeBundlesWithDaemonSet(toDir)
+	if err != nil {
+		fmt.Fprint(errLog, err.Error())
+		return
+	}
+}
+
+func (s *HarvesterServer) getSupportBundleImage() string {
+	defaultImage := "harvester/support-bundle-utils:latest"
+	clientset, err := versioned.NewForConfig(s.RESTConfig)
+	if err != nil {
+		logrus.Errorf("fail to get rest clientset: %s", err)
+		return defaultImage
+	}
+	h := NewHarvesterClient(s.Context, s.namespace, clientset)
+	setting, err := h.GetSettingByName("support-bundle-utils-image")
+	if err != nil {
+		logrus.Errorf("fail to get support-bundle-utils image from settings: %s", err)
+		return defaultImage
+	}
+	if setting.Value != "" {
+		return setting.Value
+	}
+	if setting.Default != "" {
+		return setting.Default
+	}
+	return defaultImage
+}
+
+func (s *HarvesterServer) collectNodeBundlesWithDaemonSet(toDir string) error {
+	utilsImage := s.getSupportBundleImage()
+	logrus.Debugf("support bundle utils image: %s", utilsImage)
+
+	c := NewK8sClient(s.Context, s.namespace, s.ClientSet)
+	dsName := s.sb.getDaemonSetName()
+	daemonSet := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dsName,
+			Namespace: s.namespace,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"harvester.cattle.io/support-bundle": dsName,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"harvester.cattle.io/support-bundle": dsName,
+					},
+				},
+				Spec: corev1.PodSpec{
```

Comment:
Add tolerations to the daemonset so that pods can be scheduled to nodes under maintenance(https://github.com/rancher/harvester/blob/95a5828aaf1007068e054cdab30d1ccb8b52c715/pkg/api/node/formatter.go#L78)

Reply:
[Fixed in the manager code](https://github.com/harvester/support-bundle-utils/pull/1/commits/3b7110886b29319207ef50c74670f6892cdd7492#diff-df9968bf3ccabbe861d3e72f47323abf9cb421f8e238b1a4506ef8a58c1646d5R65-R72)